src/attribute_converter.h

```diff
@@ -0,0 +1,47 @@
+/* Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MIXERCLIENT_ATTRIBUTE_CONVERTER_H
+#define MIXERCLIENT_ATTRIBUTE_CONVERTER_H
+
+#include "include/client.h"
+
+namespace istio {
+namespace mixer_client {
+
+// Underlying stream ID.
+typedef int64_t StreamID;
+
+// An interface to convert from struct to protobuf.
```

Comment:
The tricky part is: only stream_transport knows if the existing grpc stream has been closed or not.  If a stream is closed, we need to create a new attribute context.   If we put attribute_context in transport,  we can only do:   ctx->FillProto(attr, pb),  and stream_transport->Call(pb).   But the latter may say, sorry, the old grpc stream is closed, you have to re-generate protobuf with a new context.
With this interface, I can call FillProto inside StreamTranport::Call() after it determines which stream to use.